Sources/Controllers/DashboardOnMap/MapSettings/OAMapSettingsMainScreen.mm

```diff
@@ -161,19 +161,21 @@ - (void)setupView
     }];
 
     const auto resource = _app.resourcesManager->getResource(QString::fromNSString(_app.data.lastMapSource.resourceId).remove(QStringLiteral(".sqlitedb")));
-    _routesParameters = !([_app.data.lastMapSource.type isEqualToString:@"sqlitedb"] || (resource != nullptr && resource->type == OsmAnd::ResourcesManager::ResourceType::OnlineTileSources)) ? [_styleSettings getParameters:@"routes" sorted:NO] : [NSArray array];
+    _routesParameters = !([_app.data.lastMapSource.type isEqualToString:@"sqlitedb"] || (resource != nullptr && resource->type == OsmAnd::ResourcesManager::ResourceType::OnlineTileSources)) ? [_styleSettings getParameters:ROUTES_CATEGORY sorted:NO] : [NSArray array];
     if (_routesParameters.count > 0)
     {
         NSArray<NSString *> *orderedNames = @[SHOW_CYCLE_ROUTES_ATTR, SHOW_MTB_ROUTES_ATTR, HIKING_ROUTES_OSMC_ATTR, ALPINE_HIKING_ATTR, PISTE_ROUTES_ATTR, HORSE_ROUTES_ATTR, WHITE_WATER_SPORTS_ATTR, TRAVEL_ROUTES];
         _routesParameters = [_routesParameters sortedArrayUsingComparator:^NSComparisonResult(OAMapStyleParameter *obj1, OAMapStyleParameter *obj2) {
             NSInteger orderValue1 = [orderedNames indexOfObject:obj1.name];
             NSInteger orderValue2 = [orderedNames indexOfObject:obj2.name];
-            if (orderValue1 < orderValue2)
-                return NSOrderedAscending;
-            else if (orderValue1 > orderValue2)
-                return NSOrderedDescending;
-            else
-                return NSOrderedSame;
+            if (orderValue1 != NSNotFound && orderValue2 != NSNotFound)
+            {
+                if (orderValue1 < orderValue2)
```

Comment:
The second part of the comment was ignored. You should compare numbers using the NSNumber's ```compareTo:``` method.